src/routes/leagues.ts

```diff
@@ -0,0 +1,60 @@
+import { Path, GET, POST, PUT, DELETE, PathParam, Errors } from 'typescript-rest'
+import { Inject } from 'typescript-ioc'
+import Database from '../services/database'
+import League from '../models/league.model'
+import { ILeague } from '../types/models.d'
+
+@Path('/api/leagues')
+export default class LeaguesController {
+  @Inject
+  private database: Database
+
+  @GET
+  async getLeagues(): Promise<ILeague[]> {
+    return await this.database.models.League.findAll({})
+  }
+
+  @GET
+  @Path(':id')
+  async getLeague(@PathParam('id') leagueId: number): Promise<ILeague> {
+    try {
+      const league = await this.database.models.League.findById(leagueId)
+
+      if (!league) {
+        throw new Error('not found')
+      }
+
+      return league
+    } catch (e) {
+      throw new Errors.NotFoundError('League not found.')
```

Comment:
Možná by bylo lepší: throw new Errors.NotFoundError('League not found. ' + e) ??

Reply:
To by nebylo, protoze tohle prijde primo na klienta, to by bylo bezpecnostni riziko. Spis by se ta chyba mela nekam logovat. Tzn. doplnit logovani :)